File: Algorithms/Strings/HackerRank in a String!.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

class Result
{

    /*
     * Complete the 'hackerrankInString' function below.
     *
     * The function is expected to return a STRING.
     * The function accepts STRING s as parameter.
     */

    public static string hackerrankInString(string s)
    {
        string str = "hackerrank";
        int i = 0;
        int j = 0;
        while (j < s.Length && i < str.Length)
        {
            if (s[j] == str[i])
            {
                i++;
            }
            j++;
        }
        if (i == str.Length)
        {
            return "YES";
        }
        else
        {
            return "NO";
        }
    }
}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int q = Convert.ToInt32(Console.ReadLine().Trim());

        for (int qItr = 0; qItr < q; qItr++)
        {
            string s = Console.ReadLine();

            string result = Result.hackerrankInString(s);

            textWriter.WriteLine(result);
        }

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Algorithms/Strings/HackerRank in a String!.php
<?php

/*
 * Complete the 'hackerrankInString' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts STRING s as parameter.
 */

function hackerrankInString($s) {
    # Write your code here
    $hackerrank = "hackerrank";
    $i = 0;
    $chars = str_split($s);
    foreach ($chars as $c) {
        if ($c == $hackerrank[$i]) {
            $i++;
            if ($i == strlen($hackerrank)) {
                return "YES";
            }
        }
    }
    return "NO";
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$q = intval(trim(fgets(STDIN)));

for ($q_itr = 0; $q_itr < $q; $q_itr++) {
    $s = rtrim(fgets(STDIN), "\r\n");

    $result = hackerrankInString($s);

    fwrite($fptr, $result . "\n");
}

fclose($fptr);
